sdk/core/azure-core/src/main/java/com/azure/core/util/serializer/JsonUtils.java

```diff
@@ -283,76 +282,71 @@ public static <T> T getNullableProperty(JsonReader jsonReader, Function<JsonRead
-     * {@link JsonToken#NULL} when this method is called.
-     */
-    public static void readFields(JsonReader jsonReader, Consumer<String> fieldNameConsumer) {
-        readFields(jsonReader, false, fieldName -> {
-            fieldNameConsumer.accept(fieldName);
-            return false;
-        });
-    }
-
-    /**
-     * Reads the fields of a JSON object until the end of the object is reached.
-     * <p>
-     * The passed {@link JsonReader} will point to the field value each time {@code fieldNameConsumer} is called.
-     * <p>
-     * An {@link IllegalStateException} will be thrown if the {@link JsonReader#currentToken()} isn't
-     * {@link JsonToken#START_OBJECT} or {@link JsonToken#NULL} when the method is called.
+     * Reads and returns the current JSON object the {@link JsonReader} is pointing to. This will mutate the current
+     * location of {@code jsonReader}.
      * <p>
-     * If {@code readAdditionalProperties} is true and {@code fieldNameConsumer} returns false the JSON field value
-     * will be read as if it were an additional property. After the object completes reading the untyped additional
-     * properties mapping will be returned, this may be null if there were no additional properties in the JSON object.
+     * If the {@code jsonReader} is pointing to {@link JsonToken#NULL} null will be returned. Otherwise, the current
+     * JSON object will be read until completion and returned as a raw JSON string.
      *
      * @param jsonReader The {@link JsonReader} being read.
-     * @param readAdditionalProperties Whether additional properties should be read.
-     * @param fieldNameConsumer The field name consumer function.
-     * @return The additional property map if {@code readAdditionalProperties} is true and there were additional
-     * properties in the JSON object, otherwise null.
-     * @throws IllegalStateException If {@link JsonReader#currentToken()} isn't {@link JsonToken#START_OBJECT} or
-     * {@link JsonToken#NULL} when this method is called.
+     * @return The buffered JSON object the {@link JsonReader} was pointing to, or null if it was pointing to
+     * {@link JsonToken#NULL}.
+     * @throws IllegalStateException If the {@code jsonReader}'s {@link JsonReader#currentToken() current token} isn't
+     * one of {@link JsonToken#NULL}, {@link JsonToken#START_OBJECT}, {@link JsonToken#START_ARRAY}, or
+     * {@link JsonToken#FIELD_NAME}.
      */
-    public static Map<String, Object> readFields(JsonReader jsonReader, boolean readAdditionalProperties,
-        Function<String, Boolean> fieldNameConsumer) {
-        if (jsonReader.currentToken() != JsonToken.START_OBJECT && jsonReader.currentToken() != JsonToken.NULL) {
-            throw new IllegalStateException("Expected the current token of the JsonReader to either be "
-                + "START_OBJECT or NULL. It was: " + jsonReader.currentToken());
-        }
-
+    public static String bufferedJsonObject(JsonReader jsonReader) {
```

Comment:
Would it be possible to return `JsonReader` instead of string? This way we can optimize it later and API will remain the same.